soap/src/java/com/zimbra/soap/mail/type/FilterAction.java

```diff
@@ -749,6 +749,9 @@ public void validateDeleteheaderAction() throws ServiceException {
             if (last != null && last && offset == null) {
                 throw ServiceException.PARSE_ERROR(":index <offset> must be provided with :last", null);
             }
+            if (test == null) {
+                throw ServiceException.PARSE_ERROR("<test> is mandatory in action", null);
+            }
```

Comment:
This NULL check seems to contradict the JAXB contract that is being defined earlier in this file: @XmlElement(name=AdminConstants.E_TEST /* test */, required=false)
if "test" is required, then the annotation should state so and this NPE will be avoided without the explicit NULL check. Otherwise, the JAXB annotation is effectively invalid.

Reply:
agree. will remove null check and add required=true.